src/app.js

```diff
@@ -35,10 +35,21 @@ export class App {
 
     // Handles tabs being opened
     // If the tab belongs to a recorded container, then records the tab
-    this.handleTabCreated = (tab) => {
+    this.handleTabCreated = async (tab) => {
       const cookieStoreId = tab.cookieStoreId;
       if (this.containers.has(cookieStoreId)) {
         this.addTabToDb(tab);
+      } else if (cookieStoreId !== "firefox-default") {
+        let container = await browser.contextualIdentities.get(cookieStoreId);
+        if (container.name === "%TEMP%") {
```

Comment:
@ninevra what do you think about the whole idea of having a "special name" `%TEMP%`? Another approach I considered was to let the client generate the name that fully matches the scheme used by this extension, i.e. to reimplement `genName(container)` by the caller - the reason I didn't go in that direction was because the rules are not straightforward (not just `Temp` followed by random chars) and might change in the future.

Reply:
Having a special name doesn't strike me as precisely *elegant*, but it's simple and reliable, which is what's important.

`genName(container)` will definitely change in the future if i ever work on this extension more, so i agree asking the caller to reimplement it is not practical. (In particular, it might lengthen, switch to base64, stop incorporating `cookieStoreId`, and/or include some info about the machine the installation is on, to mitigate problems with Multi-Account Containers Sync.)

Possibly the special name should not have `%` in it? i've failed to make that work without urlencoding (`ext+container:name=%25TEMP%25&url=<url>`), which is not easy to communicate to the user.

Other options i've thought of:
1. exposing a message-passing interface to other extensions via `runtime.connect()`; they would pass a request and receive a `cookieStoreId` for a new temporary container.
    - requires the other extension actively choose to support this, which is way too much to ask, we have like 50 users lol
    - have to be careful not to remove that container too early
        - they could pass a message back to indicate the container has become collectable
        - (tbh this is a problem in general, though)
        - seems like an argument in favor of `tab.onCreated`; there we know the container was at some point nonempty.
2. expose our own version of the other extension's functionality (`ext+tempcontainer:<url>` in this case)
      - doesn't automatically support other extensions that create containers

Presently my inclination is to have a "special name" and maybe *also* option 2.

What do you think?

Reply:
I completely agree with your assessment, well written! 

Interestingly I don't have to urlencode `%` for some reason, but I don't have any stake in the precise special name - as long as it doesn't accidentally match what a real user would want to use, we should be fine :slightly_smiling_face: A tilde perhaps, i.e. `~TEMP~`?

Reply:
> Interestingly I don't have to urlencode % for some reason

Good to know. i'm struggling with "Open external links in a container" in general for some reason lol. i'll poke at it again later and see if i can figure out what i was doing wrong.

Reply:
Ok :grin: For reference or if that helps, my usage is literally [this script](https://github.com/maximbaz/dotfiles/blob/master/.local/bin/browser), it runs when I click on a url in any app, which then parses my list of firefox containers, prepends temp container to that list, and offers you a choice of where to open that clicked link. The only magic trick is to remember to urlencode the url itself, the rest just worked out of the box for me.